src/components/Auction/Auction.jsx

```diff
@@ -145,21 +152,46 @@ class Auction extends Component {
     });
   };
 
-  sellTokens = () => {
+  sellTokens = async () => {
     this.props.sellInAuctionStart();
     const { tokensForSell, account } = this.state;
 
     const nectarContract = new this.web3.eth.Contract(abis.necContract, this.nectarAddress);
+    const engineContract = await eth.getEngineContract();
+    nectarContract.methods.allowance(account, engineContract._address)
+      .call().then(a => {
+        nectarContract.methods
+        .approve('0xfCeB04C697a7708776C3936760aE256Eb92046f5', tokensForSell)
+        .send({ from: account })
+        .then(res => {
+          this.props.sellAndBurn(tokensForSell, account);
+        }).catch(e => {
+          debugger;
+        })
+      }).catch(e => {
+        debugger
```

Comment:
Just to re-iterate the logic here:

1. we will check if :
`nectarContract.methods.allowance()` > tokensForSell

2. If it is, we skip doing `nectarContract.methods.approve()`

3. If it is not, we do `nectarContract.methods.approve(engineContract.address), 2*200)`